frmts/raw/lcpdataset.cpp

```diff
@@ -811,43 +811,21 @@ GDALDataset *LCPDataset::Open(GDALOpenInfo *poOpenInfo)
 /*  calculate them by default.                                          */
 /************************************************************************/
 
-CPLErr LCPDataset::ClassifyBandData(GDALRasterBand *poBand,
-                                    GInt32 *pnNumClasses, GInt32 *panClasses)
+CPLErr LCPDataset::ClassifyBandData(GDALRasterBand *poBand, GInt32 &nNumClasses,
+                                    GInt32 *panClasses)
 {
-    if (pnNumClasses == nullptr)
-    {
-        CPLError(CE_Failure, CPLE_AppDefined, "Invalid pointer for panClasses");
-        return CE_Failure;
-    }
-
-    if (panClasses == nullptr)
-    {
-        CPLError(CE_Failure, CPLE_AppDefined, "Invalid pointer for panClasses");
-        *pnNumClasses = -1;
-        return CE_Failure;
-    }
-
-    if (poBand == nullptr)
-    {
-        CPLError(CE_Failure, CPLE_AppDefined,
-                 "Invalid band passed to ClassifyBandData()");
-        *pnNumClasses = -1;
-        memset(panClasses, 0, 400);
-        return CE_Failure;
-    }
+    CPLAssert(poBand);
+    CPLAssert(panClasses);
 
     const int nXSize = poBand->GetXSize();
     const int nYSize = poBand->GetYSize();
-    double dfMax = 0.0;
-    double dfDummy = 0.0;
-    poBand->GetStatistics(FALSE, TRUE, &dfDummy, &dfMax, &dfDummy, &dfDummy);
 
-    const int nSpan = static_cast<int>(dfMax);
     GInt16 *panValues =
         static_cast<GInt16 *>(CPLMalloc(sizeof(GInt16) * nXSize));
-    GByte *pabyFlags =
-        static_cast<GByte *>(CPLMalloc(sizeof(GByte) * nSpan + 1));
-    memset(pabyFlags, 0, nSpan + 1);
+    GByte *pabyFlags = static_cast<GByte *>(CPLCalloc(1, 65536));
+
+    /* so that values in [-32768,32767] are mapped to [0,65535] in pabyFlags */
+    constexpr int OFFSET = 32768;
```

Comment:
does it make sense to use numeric_limits here?

Reply:
done